pkg/appgw/backendhttpsettings.go

```diff
@@ -169,20 +169,18 @@ func (c *appGwConfigBuilder) getBackendsAndSettingsMap(cbCtx *ConfigBuilderConte
 	for backendID, serviceBackendPairs := range serviceBackendPairsMap {
 		if len(serviceBackendPairs) > 1 {
 			// more than one possible backend port exposed through ingress
-			e := controllererrors.NewErrorf(
-				controllererrors.ErrorMultipleServiceBackendPortBinding,
-				"service:port [%s:%s] has more than one service-backend port binding",
-				backendID.serviceKey(), backendID.Backend.ServicePort.String(),
-			)
-			c.recorder.Event(backendID.Ingress, v1.EventTypeWarning, events.ReasonPortResolutionError, e.Error())
-			glog.Warning(e.Error())
-			return nil, nil, nil, e
+			glog.V(5).Infof("service:port [%s:%s] has more than one service-backend port binding", backendID.serviceKey(), backendID.Backend.ServicePort.String())
 		}
 
-		// At this point there will be only one pair
+		// in case there are multiple backend ports found, using the smallest port in http setting
+		var backendport Port
+		backendport = 65536
 		var uniquePair serviceBackendPortPair
 		for k := range serviceBackendPairs {
-			uniquePair = k
+			if k.BackendPort <= backendport {
+				uniquePair = k
```

Comment:
should we log here about how we are making the decision ?

Reply:
we should also raise an k8s event on the ingress object on which we found the problem. You can find ample examples in the code to raise an event.

Reply:
We used to raise an event when we encountered such situation, i will be reusing the same code.